File: Back/src/main/java/com/gestion/adhesion/models/Adherent.java
package com.gestion.adhesion.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@EqualsAndHashCode(of = {"id"})
@Entity
@Table(	name = "adherents",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "id"),
        })
@JsonIgnoreProperties(ignoreUnknown = true)
public class Adherent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String prenom;

    private String nom;

    private String genre;

    private String email;

    private String telephone;

    private LocalDate naissance;

    private String lieuNaissance;

    private boolean referent = false;

    private String adresse;

    private boolean adresseReferent = true;

    private boolean telephoneReferent = true;

    private boolean emailReferent = true;

    private boolean mineur = false;

    private String nomLegal;

    private String prenomLegal;

    private boolean legalReferent = true;

    private boolean completReferent = false;

    private boolean completAdhesion = false;


    @OneToMany(cascade = CascadeType.ALL)
    private List<Accord> accords = new ArrayList<>();

    @OneToMany
    private List<Document> documents = new ArrayList<>();

    @ManyToOne
    private Tribu tribu;

    @OneToMany(mappedBy="adherent", cascade = CascadeType.ALL)
    private Set<Adhesion> adhesions = new HashSet<>();

    @ManyToMany
    @JsonIgnore
    private Set<Activite> cours = new HashSet<>();

    @OneToOne
    @JsonIgnoreProperties({"adherent","tokens" })
    private User user;
}

File: Front/src/app/models/adherent.ts
import { Accord } from "./accord";
import { Activite } from "./activite";
import { Adhesion } from "./adhesion";
import { Document } from "./document";
import { Tribu } from "./tribu";
import { User } from "./user";

export class Adherent {
  id: number;
  prenom: string;
  nom: string;
  genre: string;
  email: string;
  telephone: string;
  adresse: string;
  naissance: Date | undefined;
  lieuNaissance: string;
  referent: boolean;
  cours: Activite[];
  emailReferent: boolean;
  telephoneReferent: boolean;
  adresseReferent: boolean;
  nomLegal: string;
  prenomLegal: string;
  legalReferent: boolean;
  mineur: boolean;
  tribu: Tribu;
  tribuId: number
  editMail: boolean
  adhesions: Adhesion[]
  rgpdDate: Date | undefined;
  completReferent: boolean;
  completAdhesion: boolean;
  certifDate: Date | undefined;
  accords: Accord[];
  user: User | null;
  documents: Document[];

  constructor(id:number) {
    this.id = id;
    this.prenom = "";
    this.nom = "";
    this.genre = "";
    this.cours = [];
    this.email = "";
    this.telephone = "";
    this.adresse = "";
    this.lieuNaissance = "";
    this.referent = false;
    this.emailReferent = true;
    this.telephoneReferent = true;
    this.adresseReferent = true;
    this.nomLegal = "";
    this.prenomLegal = "";
    this.legalReferent = true;
    this.mineur = true;
    this.adhesions = [];
    this.completReferent = false;
    this.completAdhesion = false;
    this.accords = [];
    this.tribu = new Tribu
    this.tribuId = 0
    this.editMail = false
    this.user = null;
    this.documents = []
  }



}
